glean-core/src/debug.rs

```diff
@@ -128,17 +148,41 @@ where
     }
 }
 
-/// The debug view tag is the value for the `X-Debug-ID` header of tagged ping requests,
-/// thus is it must be a valid header value.
+fn get_bool_from_str(value: String) -> Option<bool> {
+    std::str::FromStr::from_str(&value).ok()
+}
+
+fn identity(value: String) -> Option<String> {
+    Some(value)
+}
+
+fn tokenize_string(value: String) -> Option<Vec<String>> {
+    let trimmed = value.trim();
+    if trimmed.is_empty() {
+        return None;
+    }
+
+    Some(
+        trimmed
+            .split(',')
+            .map(|s| s.trim().to_string())
+            // Filter out tags starting with "glean". They are reserved.
+            .filter(|s| !s.starts_with("glean"))
```

Comment:
*Maybe* silently removing these will bite us later.  Maybe we should at least log when this happens, or better yet return an error. Since this is part of debugging API, I don't know if the "never throw an exception out to the user" rule should apply.

Reply:
Makes sense. I made tokenize now be only responsible of the extraction. The error will be printed in the validation phase below.